### ISSUE TITLE
[API Wrapper] Support TypeScript 

### ISSUE THREAD

**Issue by dimitryzub** [serpapi/public-roadmap#278]
This issue is to track the number of current or potential user requests for adding support for TypeScript.
___
I've received a [potential request on Stackoverflow comment about using our API but in TypeScript](https://stackoverflow.com/questions/56141694/free-api-library-that-allows-processing-iterating-querying-scientific-articles/71237746?noredirect=1#comment129783097_71237746).

Request(s) count: 1

**Comment by ilyazub** [comment]
Our SDK for Node.js works without issues for TypeScript.

https://replit.com/@ilyazub/scrape-google-search-serpapi-typescript#index.ts

**Comment by ilyazub** [comment]
Our SDK for Node.js works without issues for TypeScript.

https://replit.com/@ilyazub/scrape-google-search-serpapi-typescript#index.ts

**Comment by ilyazub** [comment]
Our regular `google-search-results-nodejs` works with TypeScript too: https://replit.com/@ilyazub/scrape-google-search-serpapi-typescript#src/index.ts

We don't provide type declarations but it still works fine.

**Comment by marm123** [comment]
Yes, this is true @dimitryzub, we received such a request on Friday.

**Comment by dimitryzub** [comment]
@ilyazub Should we mention about it explicitly in our [Node integration](https://serpapi.com/integrations/node) that it works with TypeScript with no problems?

**Comment by hartator** [comment]
Maybe we can release a slightly-modified version of `google-search-results-nodejs` in TypeScript?

Regarding type contains, I remember @jvmvik had a same issue with Go. He ended up using some sort of "DynamicType" type, `interface{}`. Maybe there is something similar in TypeScript.

Let's also take chance to release as `searpapi-typescript` for the repo and lib name if we do that.

**Comment by dimitryzub** [comment]
@marm123 When you have time, could you please edit my initial issue and add a link to a potential or direct user request so we can back up this decision with data? 💛

**Comment by hartator** [comment]
> @marm123 When you have time, could you please edit my initial issue comment and add a link to a potential or direct user request so we can back up this decision with data (requests count)? 💛

> Yes, this is true @dimitryzub, we received such a request on Friday.

Yes, can you add the Intercom thread @marm123? Thank you.

**Comment by marm123** [comment]
@dimitryzub @hartator @ilyazub Really sorry I've missed this conversation! The notification got lost in my email mailbox. Just in case this is still needed here is the intercom conversation:
<a href="https://app.intercom.com/a/apps/yzck75jr/inbox/inbox/all/conversations/102910000029483">Intercom</a>

I have also added it to your initial comment @dimitryzub 

Once again, sorry for missing that.

**Comment by richardm** [comment]
I see you recently added limited support for TypesScript. It would be **very** helpful if you also provided type declarations indicating which fields I can expect to see returned from various engines, especially given that the response structure varies by engine, whether it's async or not, and probably many other variables. I've spent many hours this week reviewing the docs and reverse engineering the types from the responses I get for various searches. I expect this process will continue as I add additional use cases, and that wouldn't be necessary with better type support.

**Comment by schaferyan** [comment]
Hi Richard, thanks for your comment.

Unfortunately, we can't provide type declarations for all the fields you can expect to see returned by various engines. 

The reason for this is that some of the key names are generated dynamically based on what the search engines provide. We also can never guarantee that any particular field will always be present in the response, as this depends on whether the search engines return them.

I hope that makes sense. If you have any other questions or concerns, you are always more than welcome to email us at **support@gmail.com**, or contact us through the chat interface on our [website](https://serpapi.com/), we will be happy to help.

**Comment by schaferyan** [comment]
Another request for TypeScript support. This user is mostly interested in a list of types:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000069290?view=List)
